### PR TITLE
Clean up technical debt

### LINKED ISSUE #706: Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"
### Problem

In nonupdated browsers, not properly providing the proper `rel` attribute potentially allows the destination site to potentially hijack the browser via rogue JavaScript execution.

### Afflictions

```bash
C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\blog\labels.svelte
  18:7  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\button\button.svelte
  6:1  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\blog\[id]\+page.svelte
  52:21  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\quiz\more-info.svelte
  44:13  warning  Security: Anchor with "target=_blank" should have rel attribute containing the value "noreferrer"  security-anchor-rel-noreferrer
```

### References

- <https://stackoverflow.com/a/50709724>

### LINKED ISSUE #707: A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event
### What happened?

Accessibility standards decree that visible, non-interactive elements with an on:click event must be accompanied by an `on:keydown`, `on:keyup`, or `on:keypress` event.

### Affected code

```bash
C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\nav\bar.svelte
  30:3  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\lib\components\select\select.svelte
  21:5  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
  25:9  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events

C:\Users\Ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\quiz\quiz.svelte
  143:7  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
  158:9  warning  A11y: visible, non-interactive elements with an on:click event must be accompanied by an on:keydown, on:keyup, or on:keypress event  a11y-click-events-have-key-events
```

### LINKED ISSUE #762: "change package dot json to 4 point 0"

### LINKED ISSUE #772: `$app/navigation`'s `prefetch` is deprecated
### What happened?

```
c:\Users\ethan\Documents\GitHub\acmcsuf.com\src\routes\(site)\+layout.svelte:2:12
Error: Module '"$app/navigation"' has no exported member 'prefetch'. (ts)
<script lang="ts">
  import { prefetch } from '$app/navigation';
  import { onMount } from 'svelte';


====================================
svelte-check found 1 error and 0 warnings
```

We used this mechanism in an attempt to preload the events page from whatever page the user lands on. It seems to have been deprecated in favor of [`preloadData`](https://kit.svelte.dev/docs/modules#$app-navigation-preloaddata).

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
### Changelog

- Updated all NPM packages.
- Resolved the following issues.

### Resolved issues

- #706
- #707
- #762
- #772

### CODE REVIEW

**Review by k4rni — APPROVED** [review]
Thank you for the PR! Love to see our website moving up to its next best version:)

File: package.json [hunk 1]
```diff
@@ -27,34 +27,33 @@
...
-    "@typescript-eslint/parser": "^5.42.0",
-    "@vitest/coverage-c8": "^0.24.4",
-    "devalue": "^4.2.0",
-    "discord.js": "^13.6.0",
+    "@types/rss": "^0.0.30",
+    "@typescript-eslint/eslint-plugin": "^5.52.0",
+    "@typescript-eslint/parser": "^5.52.0",
+    "@vitest/coverage-c8": "^0.28.5",
+    "devalue": "^4.3.0",
+    "discord.js": "^14.7.1",
```
Comment (EthanThatOneKid) [inline package.json:41]: For legacy purposes, the Discord.js dependency should remain at `v^13.6.0`.

Comment (EthanThatOneKid) [reply package.json:41]: Attempted to resolve via `npm i -D discord.js@^13.6.0`.
 <https://github.com/EthanThatOneKid/acmcsuf.com/blob/b5e94546b96632b2955ac52838a323ca09b5a47d/package.json#L41>

Comment (EthanThatOneKid) [reply package.json:41]: Closing #582 proved that `scripts/close-issue-channel.js` is still working. 

image

To clean up our legacy open issue channels (pictured above), we plan to close and reopen all of the issues such that any existing conversations get archived as intended via `scripts/close-issue-channel.js` and each thread gets reopened as a new issue thread. To prepare for this, I created <https://github.com/EthanThatOneKid/gitcord/pull/23>.

**Pull request merged**